# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: mozglue/baseprofiler/core/shared-libraries-linux.cc
# Commit: c447ddfd70a2
# Full Hash: c447ddfd70a2865af1d7899a7586023053031f48
# Author: Nazım Can Altınova <canaltinova@gmail.com>
# Date: 2024-11-26 09:36:10
# Regressor Bug: 1634785
# File Overlap Count: 1
# Description:
#   Bug 1634785 - Fix how shared library names are found on linux and macOS r=mstange,profiler-reviewers
#   
#   While testing the de-duplication on Linux and macOS, I noticed that the
#   symbolication was completely broken due to having incorrect library names in
#   libs array. They were always like: "<obj-dir>/dist/bin/libxul.so" instead of
# ==============================================================================

diff -r 34ae05420bc4 -r c447ddfd70a2 mozglue/baseprofiler/core/shared-libraries-linux.cc
--- a/mozglue/baseprofiler/core/shared-libraries-linux.cc	Mon Nov 25 22:09:02 2024 +0000
+++ b/mozglue/baseprofiler/core/shared-libraries-linux.cc	Mon Nov 25 22:09:03 2024 +0000
@@ -696,7 +696,7 @@
                                          unsigned long offset = 0) {
   std::string pathStr = path;
 
-  size_t pos = pathStr.rfind('\\');
+  size_t pos = pathStr.rfind('/');
   std::string nameStr =
       (pos != std::string::npos) ? pathStr.substr(pos + 1) : pathStr;
 